src/mono/mono/metadata/metadata.c

```diff
@@ -6451,7 +6451,7 @@ mono_metadata_properties_from_typedef (MonoImage *meta, guint32 index, guint *en
 	}
 
 	start = mono_metadata_decode_row_col (tdef, loc.result, MONO_PROPERTY_MAP_PROPERTY_LIST);
-	if (loc.result + 1 < mono_metadata_table_num_rows (meta, MONO_TABLE_PROPERTYMAP)) {
+	if (GUINT32_TO_INT(loc.result + 1) < mono_metadata_table_num_rows (meta, MONO_TABLE_PROPERTYMAP)) {
```

Comment:
this shoudl be an unsigned comparison.  `mono_metadata_table_num_rows` return type should be changed to `guint32`